MHCollapsibleViewManager.m

```diff
@@ -0,0 +1,228 @@
+    {
+        MHFilterLabel *label = [data objectAtIndex:i];
+        if(label.checked)
+            [selected addObject:label];
+    }
+    return selected;
+}
+
+//creates and returns a default accessorized cell to be modified
+- (UITableViewCell*)returnAccessorizedCell:(UITableViewCellAccessoryType)accessory tableView:(UITableView*)tableView
+{
+    NSString *cellIdentifier = [NSStringFromClass([self class]) stringByAppendingString:@"cell"];
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier];
+    if(!cell)
+        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:cellIdentifier];
+    cell.accessoryType = accessory;
+    cell.textLabel.textColor = [UIColor blackColor];
+    return cell;
+}
+
+//creates a cell and returns what it should look like depending on if the header has been clicked
+//or if normal cell has been clicked
+-(UITableViewCell*)returnCell:(NSIndexPath*)indexPath tableView:(UITableView*)tableView
+{
+    UITableViewCell *cell;
+    if(indexPath.row == 0)
+    {
+        if(self.expanded)
+            cell =  [self setCellSettings:[self returnAccessorizedCell:UITableViewCellAccessoryNone tableView:tableView] type:@"ClickedHeader"];
+        else
+            cell = [self setCellSettings:[self returnAccessorizedCell:UITableViewCellAccessoryNone tableView:tableView] type:@"NormalHeader"];
+    }
+    else
+    {
+        //-1 is for offset of header row since data does not include header
+        MHFilterLabel *label = [data objectAtIndex:indexPath.row-1];
+        if(label.checked)
+            cell = [self setCellSettings:[self returnAccessorizedCell:UITableViewCellAccessoryNone tableView:tableView] type:@"ClickedCell"];
+        else
+            cell = [self setCellSettings:[self returnAccessorizedCell:UITableViewCellAccessoryNone tableView:tableView] type:@"NormalCell"];
+        cell.textLabel.text = label.name;
+    }
+    return cell;
+}
+
+- (void) selectedRowAtIndexPath: (UITableView*)tableView indexPath:(NSIndexPath*)indexPath
+{
+    //normal cell that was selected
+    if(indexPath.row > 0)
+    {
+        MHFilterLabel *label = [data objectAtIndex:indexPath.row-1];
+        NSString *type = nil;
+        
+        label.checked = !label.checked;
+        
+        if(label.checked)
+            type = @"ClickedCell";
+        else
+            type = @"NormalCell";
+        
+         [self setCellSettings:[tableView cellForRowAtIndexPath:indexPath] type:type];
+    }
+    //header cell, add data, change look, etc.
+    else
+        [self toggleCollapse:tableView indexPath:indexPath];
+}
+
+
+//toogles the children rows and handles the look of the header row
+//as a result from collapsing/expanding
+- (void) toggleCollapse: (UITableView*)tableView indexPath:(NSIndexPath*)indexPath
+{
+    //toogle expanded, used to check for if the header has been clicked
+    self.expanded = !self.expanded;
+    
+    int count = (int)self.numOfRows;
```

Comment:
This is minor but I would recommend using NSInteger or NSUInteger (for unsigned values). They do a better job of dealing with different hardware architectures. I think they can adapt the number of bits allocated based on architecture. I very rarely have a case where I need to use primitive c types. The other upside is if you use the apple macros for everything it also makes moving to swift smoother.